### ISSUE TITLE
`shouldDecompressImages` property related problem?

### ISSUE THREAD

**Issue by welsonxue** [rs/SDWebImage#1927]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

In my opinion, the property `shouldScaleImage ` or `shouldDecompressImages ` is related with single image. If we set them in SDImageCache, all the images are using this setting, maybe not satisfy our requirements. Can we move these property to `SDWebImageOptions`?

Sometimes some huge images will cause memory exhausting and even crash because they cannot choose `shouldDecompressImages` with single image. If we set this property into `SDWebImageOptions`, maybe it would be better.

**Comment by dreampiggy** [comment]
Can be considered. If we put a `BOOL` value in the context arguments in SD 5.x. Does this acceptable ? Because if we just represent it as a BOOL value, there is no way to specify that `use decompress only for this request, others disable` because BOOL represent whether `YES` or `NO`. But not contains a `default` value.

Like this:

```objective-c
FOUNDATION_EXPORT SDWebImageContextOption _Nonnull const SDWebImageContextDecompressImages;
```

Then you need to specify a `NSNumber` value, if you do not specify, use `downloader` or `cache`'s built-in property instead.

```objective-c
UIImageView *imageView;
[imageView sd_setImageWithURL:url placeholderImage:nil options:0 context:@{SDWebImageContextDecompressImages: @(YES)} progress:nil completed:nil];
```

**Comment by dreampiggy** [comment]
@welsonxue Hi. This feature has been merged in 5.x.
You can now use `SDWebImageAvoidDecodeImage` to sepcify that current image reuqest should not use force decode feature. There is no central control property any more.

But however, if you need this, you can do this by create your own category instead.

```objective-c

@interface UIImageView (MyCategory)

- (void)my_setImageWithURL:(NSURL *)url placeholderImage:(UIImage *)image options:(SDWebImageOptions)options;

@end

@implementation UIImageView (MyCategory)

- (void)my_setImageWithURL:(NSURL *)url placeholderImage:(UIImage *)image options:(SDWebImageOptions)options {
    if (global_disable_force_decode) {
        options |= SDWebImageAvoidDecodeImage;
    }
    [self sd_setImageWithURL:url placeholderImage:image options:options];
}

@end
```

**Comment by ripperhe** [comment]
@dreampiggy 欲哭无泪，我决定回退版本。。。
